nc-news/src/components/Articles.jsx

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import ArticleCard from "./ArticleCard";
+import ArticleNavBar from "./ArticleNavBar";
+
+function Articles({ topic }) {
+  const [loading, setLoading] = useState(true);
+  const [articles, setArticles] = useState([]);
+
+  let request = "";
+
+  if (topic === undefined) {
+    request = `https://be-nc-news-zmuo.onrender.com/api/articles`;
+  } else {
+    request = `https://be-nc-news-zmuo.onrender.com/api/articles?topic=${topic}`;
+  }
+
+  useEffect(() => {
+    axios
+      .get(`${request}`)
+      .then((response) => {
+        setArticles(response.data.articles);
+        setLoading(false);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, [topic]);
```

Comment:
It's really good to see this component being used for both cases of either getting all articles or articles by topic 🏆 

You could also add some `loading` state and `error` state into this request to improve on the user experience